src/main/app-menu.ts

```diff
@@ -896,3 +919,7 @@ const registeredMachines = [
     displayName: "ZX Spectrum 128K"
   },
 ]
+
+function filterVisibleItems<T extends (MenuItemConstructorOptions | MenuItem)>(items: T[]): T[] {
+  return items.filter(i => i.visible !== false);
```

Comment:
visible attribute is ignored for separator items. Who knows for which else item types it is ignored as well, so we just filter the non-visible items out.